From: Yocto Build System <noreply@yoctoproject.org>
Date: Mon, 3 Nov 2025 21:53:00 +0000
Subject: [PATCH] Fix dangling reference warnings in fleet provisioning sample

The fleet provisioning sample code contains dangling reference warnings
that cause compilation failures when using newer GCC versions with -Werror.

The issue occurs when binding references to temporary objects that are
destroyed at the end of the expression. This fix changes the references
to values to avoid the dangling reference issue.

Upstream-Status: Submitted [https://github.com/aws/aws-iot-device-sdk-cpp-v2/issues/839]
Signed-off-by: Yocto Build System <noreply@yoctoproject.org>

--- a/main.cpp
+++ b/main.cpp
@@ -129,7 +129,7 @@ int main(int argc, char *argv[])
         [&createKeysResultPromise](CreateKeysAndCertificateResult &&result)
         { createKeysResultPromise.set_value(std::move(result)); });
 
-    const auto &createKeysResult = createKeysResultPromise.get_future().get().value();
+    const auto createKeysResult = createKeysResultPromise.get_future().get().value();
     if (!createKeysResult.IsSuccess())
     {
         s_onServiceError(createKeysResult.GetError(), "create-keys-and-certificate");
@@ -169,7 +169,7 @@ int main(int argc, char *argv[])
         [&registerThingResultPromise](RegisterThingResult &&result)
         { registerThingResultPromise.set_value(std::move(result)); });
 
-    const auto &registerThingResult = registerThingResultPromise.get_future().get().value();
+    const auto registerThingResult = registerThingResultPromise.get_future().get().value();
     if (!registerThingResult.IsSuccess())
     {
         s_onServiceError(registerThingResult.GetError(), "register-thing");
